contracts/TokenConverter/AbstractTokenConverter.sol

```diff
@@ -0,0 +1,787 @@
+        uint256 amountOut
+    ) internal virtual {}
+
+    /// @param accessControlManager_ Access control manager contract address
+    /// @param priceOracle_ Resilient oracle address
+    /// @param destinationAddress_  Address at all incoming tokens will transferred to
+    function __AbstractTokenConverter_init(
+        address accessControlManager_,
+        ResilientOracle priceOracle_,
+        address destinationAddress_
+    ) internal onlyInitializing {
+        __AccessControlled_init(accessControlManager_);
+        __ReentrancyGuard_init();
+        __AbstractTokenConverter_init_unchained(priceOracle_, destinationAddress_);
+    }
+
+    /// @param priceOracle_ Resilient oracle address
+    /// @param destinationAddress_  Address at all incoming tokens will transferred to
+    function __AbstractTokenConverter_init_unchained(ResilientOracle priceOracle_, address destinationAddress_)
+        internal
+        onlyInitializing
+    {
+        _setPriceOracle(priceOracle_);
+        _setDestination(destinationAddress_);
+        conversionPaused = false;
+    }
+
+    /// @notice To get the amount of tokenAddressOut tokens sender could receive on providing amountInMantissa tokens of tokenAddressIn
+    /// @dev This function retrieves values without altering token prices.
+    /// @param amountInMantissa Amount of tokenAddressIn
+    /// @param tokenAddressIn Address of the token to convert
+    /// @param tokenAddressOut Address of the token to get after conversion
+    /// @return amountConvertedMantissa Amount of tokenAddressIn should be transferred after conversion
+    /// @return amountOutMantissa Amount of the tokenAddressOut sender should receive after conversion
+    /// @custom:error InsufficientInputAmount error is thrown when given input amount is zero
+    /// @custom:error ConversionConfigNotEnabled is thrown when conversion is disabled or config does not exist for given pair
+    function _getAmountOut(
+        uint256 amountInMantissa,
+        address tokenAddressIn,
+        address tokenAddressOut
+    )
+        internal
+        view
+        returns (
+            uint256 amountConvertedMantissa,
+            uint256 amountOutMantissa,
+            uint256 tokenInToOutConversion
+        )
+    {
+        if (amountInMantissa == 0) {
+            revert InsufficientInputAmount();
+        }
+
+        ConversionConfig memory configuration = conversionConfigurations[tokenAddressIn][tokenAddressOut];
+
+        if (!configuration.enabled) {
+            revert ConversionConfigNotEnabled();
+        }
+
+        uint256 tokenInUnderlyingPrice = priceOracle.getPrice(tokenAddressIn);
+        uint256 tokenOutUnderlyingPrice = priceOracle.getPrice(tokenAddressOut);
+
+        /// amount of tokenAddressOut after including incentive
+        uint256 conversionWithIncentive = MANTISSA_ONE + configuration.incentive;
+        /// conversion rate after considering incentive(conversionWithIncentive)
```

Comment:
The comments are quite misleading here

Reply:
Done https://github.com/VenusProtocol/protocol-reserve/pull/9/commits/8d8f8c786e598bf59b0d90edda59f8df16be84af